collect_app/src/test/java/org/odk/collect/android/widgets/BarcodeWidgetTest.java

```diff
@@ -183,6 +185,23 @@ public void clickingBarcodeButton_whenFrontCameraIsAvailable_launchesCorrectInte
         assertThat(shadowActivity.getNextStartedActivity().getBooleanExtra(Appearances.FRONT, false), is(true));
     }
 
+    @Test
+    public void whenPromptHasNoAnswerAppearance_answerIsNotDisplayed() {
+        FormEntryPrompt prompt = new MockFormEntryPromptBuilder(promptWithAppearance(Appearances.HIDDEN_ANSWER))
+                .withAnswer(new StringData("original contents"))
+                .build();
+
+        BarcodeWidget widget = createWidget(prompt);
+        assertThat(widget.binding.barcodeAnswerText.getVisibility(), equalTo(View.GONE));
```

Comment:
It would be good to add a comment that this first set of asserts is for the existing answer and the one below is for a new answer. Or maybe splitting it into two tests but I think the first option is ok.